services/report/report_builder.py

```diff
@@ -84,21 +89,33 @@ def output_report(self) -> Report:
         Returns:
             Report: The legacy report desired
         """
-        if self._present_labels:
-            if self._present_labels == {
-                SpecialLabelsEnum.CODECOV_ALL_LABELS_PLACEHOLDER
-            }:
-                log.warning(
-                    "Report only has SpecialLabels. Might indicate it was not generated with contexts"
-                )
-            for file in self._report:
-                for line_number, line in file.lines:
-                    self._possibly_modify_line_to_account_for_special_labels(
-                        file, line_number, line
+        if self.should_use_label_index:
+            if len(self.label_index) > 0:
+                if len(self.label_index) == 1 and self.label_index.values() == [
+                    SpecialLabelsEnum.CODECOV_ALL_LABELS_PLACEHOLDER
+                ]:
+                    log.warning(
+                        "Report only has SpecialLabels. Might indicate it was not generated with contexts"
+                    )
+                self._report._totals = None
+                self._report.set_label_idx(self.label_index)
+        else:
+            if self._present_labels:
+                if self._present_labels and self._present_labels == {
+                    SpecialLabelsEnum.CODECOV_ALL_LABELS_PLACEHOLDER
+                }:
+                    log.warning(
+                        "Report only has SpecialLabels. Might indicate it was not generated with contexts"
                     )
-            self._report._totals = None
+                for file in self._report:
+                    for line_number, line in file.lines:
+                        self._possibly_modify_line_to_account_for_special_labels(
+                            file, line_number, line
+                        )
+                self._report._totals = None
         return self._report
 
+    # TODO: This can be removed after label indexing is rolled out for all customers
```

Comment:
can you leave a github issue number in the comment so we can grep for it and actually make sure we clean everything up

Reply:
THat's a fine idea. I'll do that.